olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -452,6 +453,20 @@ client::CancellableFuture<DataResponse> VersionedLayerClientImpl::GetData(
                                                  std::move(promise));
 }
 
+bool VersionedLayerClientImpl::RemoveFromCache(std::string partition_id) {
+  repository::PartitionsCacheRepository cache_repository(catalog_,
+                                                         settings_.cache);
+  auto partition = cache_repository.ClearPartitionMetadata(
+      catalog_version_.load(), partition_id, layer_id_);
+  if (!partition) {
+    return false;
+  }
+
+  repository::DataCacheRepository data_repository(catalog_, settings_.cache);
+  data_repository.Clear(layer_id_, partition.get().GetDataHandle());
```

Comment:
Updated, but there is an issue. First of all we are trying to get metadata from cache. `DefaultCache::Get` returns `boost::any()` in case its not opened, same as for non existing key case.
So the method will return `true` but cache is actually not valid.